### PR TITLE
Add Mark field

### LINKED ISSUE #22: Add Mark field
Add mark information to the device model.
this field need to be sent to netcontrol each time it's value is changing.

### PR DESCRIPTION
**Pull request by KwikKill**
## Description

Add mark information on devices :
- Marks can be defined in a config file before starting the gate
- Marks can be seen and edited in the front admin panel
image
image
- Marks are given to devices based of a distribution
- Marks can also be given to devices based of user's game
- This repartition can be edited in the config file or in the front admin panel
image

## Checklist

- [X] I have tested the changes locally and they work as expected.
- [X] I have tested the responsiveness of the changes and they work as expected.
- [X] I have assigned the pull request to the appropriate reviewer(s).
- [X] I have added labels to the pull request, if necessary.

## Related Issues

Fix #22

### CODE REVIEW

File: backend/assets/misc/settings.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    {
+      "name": "vpn3",
+      "value": 103,
+      "priority": 0.7
+    }
+  ],
+  "games": {
+    "csgo": 101,
+    "tm": 102,
+    "lol": 103
```
Comment (SkytAsul) [inline backend/assets/misc/settings.json:27]: I'd rather have a list here with the available mark_s_ for the game.

Comment (KwikKill) [reply backend/assets/misc/settings.json:27]: :ok_hand:
If we set the marks per game to be a list, how do you want the priority to be set (equiprobable ? list with given priority (`{ "csgo": [{101:0.3, 102:0.5, 103:0.2}],...` ? or based on priority given in the mark settings ?)

Comment (SkytAsul) [reply backend/assets/misc/settings.json:27]: Based on the priority given in the mark settings. You assign the mark in the subset given.

Comment (KwikKill) [reply backend/assets/misc/settings.json]: Done

**Review by SkytAsul — APPROVED** [review]
Only checked the backend but OK for me

File: backend/langate/network/apps.py [hunk 2]
```diff
@@ -47,43 +48,40 @@ def ready(self):
             logger.info(_("[PortalConfig] Adding previously connected devices to the ipset"))
 
             for dev in Device.objects.all():
-                connect_res = netcontrol.query("connect_user", {"mac": dev.mac, "name": dev.name})
+                userdevice = UserDevice.objects.filter(mac=dev.mac).first()
+                if userdevice is not None:
+                    connect_res = netcontrol.query("connect_user", {"mac": userdevice.mac, "name": userdevice.name})
+                else:
+                    connect_res = netcontrol.query("connect_user", {"mac": dev.mac, "name": dev.name})
```
[the remark this answers is not in the document]
Comment (KwikKill) [reply backend/langate/network/apps.py:55]: Cause that's a typo, it's supposed to be userdevice.user.username to be able to link the device with the user in netcontrol.

**Pull request merged**